ctsit/tt_engine
# add sftp capabilities to Dockerfile

Addresses #32 

Note: this PR removes the `openxlsx` library from the Docker image; the scripts that use it will no longer work until the library is removed.

To test:
1.  rebuild from the Dockerfile `docker build -t tt_engine .`
1. Create a passphraseless ssh key as described in #32, store it in `~/.ssh/junk/oneuftesting`
1. Run the following script to mount your ssh key as a volume - alter the `.env` file if needed:
```bash
docker run -v /Users/$USER/.ssh/junk/oneuftesting:/root/.ssh/id_rsa --rm --env-file .env tt_engine Rscript load_person_data_into_results.R dev_mode=1
```

___
While this works perfectly fine with a different server hosting redcap (tested with `redcap-phone`, it's not working for me using my local (docker) redcap, getting data from the sftp server works fine, but the data is not sent to the local REDCap. The issue is likely because `localhost/api` is my `URI` and the `tt_engine` container thinks `localhost` belongs to itself.

See [this stackoverflow thread](https://stackoverflow.com/questions/47017801/docker-curl-to-other-container-in-same-host) for some insight but no easy answer. redcap-docker-compose is using a network called `<REDCAP_PREFIX>_redcap_network`

Solution for local testing should look something like this:
```bash
docker run -ti \
    --network=redcap_redcap_network \
    -p "0.0.0.0:99":80 \
    -v /Users/$USER/.ssh/junk/oneuftesting:/root/.ssh/id_rsa \
    --rm --env-file .env \
    tt_engine Rscript load_person_data_into_results.R dev_mode=1
```

You will need to fiddle with the `-p` flag, it seems it should be `-p rocker_container_ip:rocker_container_port:local_redcap_port`
